R/add_cols.R

```diff
@@ -118,60 +118,97 @@ NULL
 }
 
 #' @name .add_date
-.add_deaths <- function(.data,
-                        onset_to_death,
-                        hosp_death_risk,
-                        non_hosp_death_risk) {
+.add_outcome <- function(.data,
+                         onset_to_death,
+                         onset_to_recovery,
+                         hosp_death_risk,
+                         non_hosp_death_risk,
+                         config) {
   infected_idx <- .data$infected == "infected"
   num_infected <- sum(infected_idx)
-  .data$deaths <- NA_real_
-  .data$deaths[infected_idx] <- .data$time[infected_idx] +
-    onset_to_death(num_infected)
+  .data$outcome <- "contact"
+  .data$outcome_time <- NA_real_
+  .data$outcome[infected_idx] <- "recovered"
+  .data$outcome_time[infected_idx] <- .data$time[infected_idx] +
+    onset_to_recovery(num_infected)
+  hosp_idx <- !is.na(.data$hospitalisation) & infected_idx
+  non_hosp_idx <- is.na(.data$hospitalisation) & infected_idx
 
-  apply_death_risk <- function(.data, risk, hosp = TRUE) {
+  # define functions used only by .add_outcome
+  time_varying_risk <- function(.data, died_idx, config) {
+    # check time-varying function
+    .check_func_req_args(config$time_varying_death_risk)
```

Comment:
On point 2, the time-varying cfr function given by the user does not need to evaluate between 0 and 1, that is one of the reasons for the normalisation step in the function. The only restriction is values should not be negative, in theory negative values would also work by increasing, rather than decreasing, the probability of death, but the current way the code is written does not allow for this scenario.